eventkit_cloud/ui/static/ui/app/actions/dataPackActions.js

```diff
@@ -88,6 +126,75 @@ export function getRuns(params, options = {}, isAuto) {
     };
 }
 
+export function getFeaturedRuns(args) {
+    return (dispatch, getState) => {
+        const { featuredRunsList } = getState();
+        if (featuredRunsList.fetching && featuredRunsList.cancelSource) {
+            // if this is not a direct request from the user, dont cancel ongoing requests
+            if (args.isAuto) {
+                return null;
+            }
+            // if there is already a request in process we need to cancel it
+            // before executing the current request
+            featuredRunsList.cancelSource.cancel('Request is no longer valid, cancelling');
+        }
+
+        const { CancelToken } = axios;
+        const source = CancelToken.source();
+
+        dispatch({ type: types.FETCHING_FEATURED_RUNS, cancelSource: source });
+
+        const params = {};
+        params.page_size = args.pageSize;
+        params.ordering = 'featured,-started_at';
```

Comment:
I think the ordering is like that so that you get featured runs back _first_, without excluding the rest of the runs. It is a slightly strange way to do it, and it might also be preferable to split those calls up, but I think this works for our purposes (plus it's probably more efficient). It's the way it's been done thus far - I just copied that code over from DataPackPage I think.

Reply:
Right, in this case you get runs ordered by featured first, you will still have ALL the runs returned, this was what we wanted for some of the other pages when sorting by featured. But from what I can tell, in this case, you dont actually care about any of the other runs, you want ONLY featured. So instead of /api/filter?ordering=featured` you could add support for `/api/filter?featured` or some variation of that. It should save time by not returning a full list of runs which end up getting filtered out in the frontend.

Reply:
Ah I get what you're saying now. Specifically in the case of the Dashboard, the way it's currently doing things is overkill. That's a good point. Do you think it might be a good optimization task for us to take a look at after this PR?

Reply:
Yea that sounds good, I guess just throw a task for that in scrumdo 👍

Reply:
I was just taking another look at this in the code, and it looks like `getRuns()` is only used in DashboardPage to get the datapacks for **My DataPacks*. For **Featured** I'm making a separate call to `getFeaturedRuns()`. So we might not need to worry about that optimization task for now.